packages/block-library/src/file/edit.native.js

```diff
@@ -110,24 +160,113 @@ export default class FileEdit extends Component {
 		);
 	}
 
-	render() {
-		const { attributes } = this.props;
-		const { href, fileName, downloadButtonText, id } = attributes;
+	getToolbarEditButton( open ) {
+		return (
+			<BlockControls>
+				<ToolbarGroup>
+					<ToolbarButton
+						title={ __( 'Edit file' ) }
+						icon={ replace }
+						onClick={ open }
+					/>
+				</ToolbarGroup>
+			</BlockControls>
+		);
+	}
 
-		if ( ! href ) {
-			return (
-				<MediaPlaceholder
-					icon={ <BlockIcon icon={ icon } /> }
-					labels={ {
-						title: __( 'File' ),
-						instructions: __( 'CHOOSE A FILE' ),
-					} }
-					onSelect={ this.onSelectFile }
-					onFocus={ this.props.onFocus }
-					allowedTypes={ [ 'other' ] }
-				/>
-			);
+	getInspectorControls(
+		{ showDownloadButton, textLinkTarget },
+		isUploadInProgress,
+		isUploadFailed
+	) {
+		const actionButtonStyle = this.props.getStylesFromColorScheme(
+			styles.actionButton,
+			styles.actionButtonDark
+		);
+
+		const isCopyUrlDisabled = isUploadFailed || isUploadInProgress;
+		const dimmedStyle = isCopyUrlDisabled && styles.disabledButton;
+		const finalButtonStyle = Object.assign(
+			{},
+			actionButtonStyle,
+			dimmedStyle
+		);
+
+		return (
+			<InspectorControls>
+				<PanelBody title={ __( 'File block settings' ) } />
+				<PanelBody>
+					<ToggleControl
+						icon={ external }
+						label={ __( 'Open in new tab' ) }
+						checked={ textLinkTarget === '_blank' }
+						onChange={ this.onChangeOpenInNewWindow }
+					/>
+					<ToggleControl
+						icon={ button }
+						label={ __( 'Show download button' ) }
+						checked={ showDownloadButton }
+						onChange={ this.onChangeDownloadButtonVisibility }
+					/>
+					<BottomSheet.Cell
+						disabled={ isCopyUrlDisabled }
+						label={
+							this.state.isUrlCopied
+								? __( 'Copied!' )
+								: __( 'Copy file URL' )
+						}
+						labelStyle={
+							this.state.isUrlCopied || finalButtonStyle
+						}
```

Comment:
Hi @etoledom can you explain how this ||-operator evaluates here. What happens if this.state.isUrlCopied is true? I am going to test this shortly but I am just asking here since you might be able to expound clearly. I am just verifying that I understand how these operators work in Javascript :)

Reply:
Sure, so basically the right hand side will be returned only if the left hand side is `false`. If the LHS is `true`, then `undefined` is returned.

This is a short for:
```js
return this.state.isUrlCopied ? undefined : finalButtonStyle;
```

The oposite is true using `&&` operator instead.

Reply:
> If the LHS is true, then undefined is returned.

I thought if the LHS of `||` was truthy, that value would be returned, so if the LHS was `true` it would return `true`. Am I misunderstanding javascript (again 😄 )?

Reply:
Wait a sec, you are right! It's not returning `undefined` but the LHS itself. This is new to me 🤔 
Thanks for clarifying @mchowning !

Reply:
Just to clarify. So if it's returning the LHS then does that mean if url is copied then the style would be set to true instead of an actual style ? That's what I'm trying to clear up.

Reply:
You are right, the style would be set to true.
I'm doubting this now myself, you know 🤣
But there are a few instances of this same technique used in styles in other parts of the project, so I'm inclined to say it's safe.

Reply:
Okay understood! LGTM 🚢